src/feedbinapi.cpp

```diff
@@ -0,0 +1,306 @@
+#include "feedbinapi.h"
+
+#include <cstdint>
+#include <curl/curl.h>
+#include <string>
+
+#include "3rd-party/json.hpp"
+#include "curlhandle.h"
+#include "json.h"
+#include "rss/feed.h"
+#include "strprintf.h"
+#include "utils.h"
+
+#define FEEDBIN_AUTHENTICATION_PATH "/v2/authentication.json"
+#define FEEDBIN_STARRED_ENTRIES_PATH "/v2/starred_entries.json"
+#define FEEDBIN_TAGGINGS_PATH "/v2/taggings.json"
+#define FEEDBIN_SUBSCRIPTIONS_PATH "/v2/subscriptions.json"
+#define FEEDBIN_UNREAD_ENTRIES_PATH "/v2/unread_entries.json"
+
+using json = nlohmann::json;
+
+namespace newsboat {
+
+FeedbinApi::FeedbinApi(ConfigContainer &c) : RemoteApi(c) {
+  const std::string http_auth_method = cfg.get_configvalue("http-auth-method");
+  if (http_auth_method == "any") {
+    // default to basic HTTP auth to prevent Newsboat from doubling up on HTTP
+    // requests, since it doesn't "guess" the correct auth type on the first
+    // try.
+    cfg.set_configvalue("http-auth-method", "basic");
+  }
```

Comment:
I copied this from another API file for a service that uses HTTP basic auth. I'm not certain whether or not it's necessary here.

Reply:
I think this is a safe default, so let's leave it as-is.